### PR TITLE
Javier: niLang: Emscripten params and optional input 

### PR DESCRIPTION
**Pull request by JavierFerrerGomez**
Now we can pass parameters to the module via the **Params** property. This params behave like the ones coming from the URL query and they can be overriden by the URL query params. Params are a key-value pair. Example:
```js
var Module = createVlkPlayerNiModule({
    Params: { OpenFile: vpkFile, /* more params here if your need to */ },
    // more module properties
}
```

Now we can also disable the input for the canvas if we want. We must pass **UseInput** boolean property at module creation. This is a mandatory field. Example:
```js
var Module = createVlkPlayerNiModule({
    UseInput: false,
    // more module properties
}
```

## Formal Notes

* RN:
* RN:
* NoRN:

## Description, Motivation and Context

* Link to issue: Fixes ?
* Link to reference doc: ?

## How Has This Been Tested?

You can test it like any other emscripten test
```
BUILD_TARGET=web-js . hat
ham pass1 && ham Test_niLang
```
Run Test\_niLang:
```
# Serve the $WORK folder (niLang's parent folder)
cd "$WORK"
simple-http-server
# Open the test case
open http://localhost:8123/niLang/bin/web-js/Test_niLang_ra.html
```

## Types Of Changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that causes existing functionality to change)
* [ ] Documentation (code docs, comments, or changes to the doc systems)
* [ ] Testing/Build (test coverage or the test/build subsystems themselves)
* [ ] Packaging (adds examples or modifies a release package)

### CODE REVIEW

File: sources/emscripten/0_niLang_shell.html [hunk 1]
```diff
@@ -181,6 +181,7 @@
 
       console.log('... createDefaultNiModule');
       var Module = createDefaultNiModule({
+        UseInput: true,
```
Comment (JavierFerrerGomez) [inline sources/emscripten/0_niLang_shell.html:184]: This is the default that makes sense for me.

File: sources/emscripten/0_niLang_pre.js [hunk 2]
```diff
@@ -791,6 +800,16 @@ var NIAPP = _moduleLib('NIAPP', {
   },
 
   GetParamsAsObject: function GetParamsAsObject(query) {
+    // first we get the parameters passed to the module
+    var moduleParams = Module["Params"];
```
Comment (JavierFerrerGomez) [inline sources/emscripten/0_niLang_pre.js:804]: I accept better name suggestions...

File: sources/emscripten/0_niLang_shell.html (hunk 1 above)
Comment (prenaux) [reply sources/emscripten/0_niLang_shell.html:184]: In the shell template yes.

File: sources/emscripten/0_niLang_pre.js (hunk 2 above)
Comment (prenaux) [reply sources/emscripten/0_niLang_pre.js:804]: Lets leave at that for now.

**Review by prenaux — APPROVED** [review]

**Pull request merged**